index.js

```diff
@@ -65,8 +64,43 @@ function MLLPServer(host, port) {
 
     });
 
-    Server.listen(PORT, HOST);
+    self.send = function (receivingHost, receivingPort, hl7Data, callback) {
+        var sendingClient = new net.connect({
+            host: receivingHost,
+            port: receivingPort
+        }, function () {
+            console.log('Sending data to ' + receivingHost + ':' + receivingPort);
+            sendingClient.write(VT + hl7Data + FS + CR);
+        });
+
+        sendingClient.on('data', function (rawAckData) {
+            console.log(receivingHost + ':' + receivingPort + ' ACKED data');
+
+            var ackData = rawAckData
+                .toString() // Buffer -> String
+                .replace(VT, '')
+                .split('\r')[1] // Ack data
+                .replace(FS, '')
+                .replace(CR, '');
+
+            callback(null, ackData);
```

Comment:
Hi @Eriatolc,

Thanks for the feedback. Before I implement the changes, do we agree that the data should be checked for "AA" vs "AE" to determine if an error should be sent?

As for sticking with the EventEmitter pattern... it seems to makes sense for the listener piece because the client is presumably subscribing for a while (waiting for inbound messages). The sender piece is more of a request-response pattern that the client isn't subscribing to.

Perhaps this justifies my implementation. Can you think of how the event subscription would work for the sending piece?

Thank you.